### PR TITLE
Add BACKUPS_MAX_COUNT config variable

### PR DESCRIPTION
**Pull request by Tremolo4**
Keeps only the newest N backups, deletes others.

I'm not sure about setting the default to 72. Maybe it should be set to an extremely large number instead.

Reasoning for this feature: Having a maximum amount instead of maximum age for backups makes a lot more sense for servers with longer activity gaps. Especially in combination with `BACKUPS_IF_IDLE=false`. For example, when you take a break for a week (i.e. longer than MAX_AGE), and then it turns out you've made a horrible mistake in your last session, you're gonna wish your backups hadn't all been deleted during the week.

### CODE REVIEW

**Comment by lloesche** [comment]
Hi @Tremolo4, thank you for your PR. The feature makes a lot of sense. That said I always try to make any change to the image fully backwards compatible. So adding a new feature that will delete people's backups must be turned off by default.

Let's use a default of `0` and encase the entire cleanup code with a `if [ "$BACKUPS_MAX_COUNT" -gt 0 ]; then` statement. That way nothing breaks for existing users no matter what their current backup settings are.

Also instead of pushd/popd I'm using a pattern of
```
cd "$BACKUPS_DIRECTORY" || fatal "Could not cd $BACKUPS_DIRECTORY"
...
...
cd -
```

Other than that I think we're good to go.

Was a bit taken aback by the shellcheck disable but the use of `xargs -I {}` is pretty clever so we should be good there.

**Comment by Tremolo4** [comment]
Thanks for the review! I agree with all of it. I pushed the changes.

> Was a bit taken aback by the shellcheck disable but the use of xargs -I {} is pretty clever so we should be good there.

The main part is [from stackoverflow](https://stackoverflow.com/questions/25785/delete-all-but-the-most-recent-x-files-in-bash/34862475#34862475) so I can't take credit for it 🤡, but there's some reasoning there for why it's okay. Another reason is that we're only parsing filenames that we created ourselves and not generally some completely unknown files.
If someone creates a filename with a newline in it in the backup directory, that'd break it. I think the absolute worst case then is it deletes the wrong backup. For example if we had files named `foo\nbar` and `foo` then it would delete `foo` when trying to delete `foo\nbar`

**Review by lloesche — CHANGES_REQUESTED** [review]

**Comment by lloesche** [comment]
> Another reason is that we're only parsing filenames that we created ourselves and not generally some completely unknown files.

Actually that's not the case. We don't know what `BACKUPS_DIRECTORY` the user has chosen. It could be on some NAS mount mixed with other files.

File: valheim-backup [hunk 1]
```diff
@@ -90,6 +90,15 @@ flush_old() {
         debug "No old backups to remove"
         return
     fi
+
+    if [ "$BACKUPS_MAX_COUNT" -gt 0 ]; then
+        info "Removing all but the newest $BACKUPS_MAX_COUNT backups"
+        cd "$BACKUPS_DIRECTORY" || fatal "Could not cd $BACKUPS_DIRECTORY"
+            # shellcheck disable=SC2010
+            ls -tp | grep -v '/$' | tail -n +$((BACKUPS_MAX_COUNT+1)) | xargs -I {} rm -v -- {}
```
[the remark this answers is not in the document]
Comment (Tremolo4) [reply valheim-backup]: That makes sense. I pushed a solution using `find` instead of `ls` in order to do this.

**Comment by Tremolo4** [comment]
I'm now using `find` and `sort` instead of `ls -t` in order to check for the filename `worlds-*.zip` as you suggested. For what it's worth, I've also made it work correctly with filenames containing newlines -- just needed some `-z` flags etc.

1. `find` matching files and prefix with timestamp using `-printf`. Also add NUL delimiter.
2. `sort` according to timestamp.
3. `cut` away timestamp.
4. Use `tail` to skip newest N files.
5. Use `xargs` to pass remaining list of files to `rm -v`.

**Comment by Tremolo4** [comment]
What do you think about adding `-maxdepth 1` for extra safety? Although `BACKUPS_MAX_AGE` doesn't currently use that either.

**Review by lloesche — APPROVED** [review]
lgtm 🚢

**Pull request merged**